subjects/cybersecurity/pentest-kit/README.md

```diff
@@ -0,0 +1,182 @@
+</center>
+
+### Introduction:
+
+Penetration testing (pen-testing) is a critical process in cybersecurity, involving the use of specialized tools to identify and exploit vulnerabilities in systems and networks. Pentest tools are essential for conducting thorough security assessments, helping to safeguard against potential threats.
+
+### Objective:
+
+Using a programming language from your choice you will create a a multi-functional tool for penetration testing, similar to popular security tools like `Nmap`, and `Dirsearch`. This project aims to give hands-on experience in understanding and developing security tools and building a practical toolkit for various pen-testing tasks.
+
+By completing this project, You will:
+
+- Gain practical experience in developing custom pentesting tools.
+- Understand the core functionalities of port scanning, directory brute-forcing, network mapping, and HTTP header analysis.
+- Learn how to build a cohesive set of utilities for penetration testing.
+- Enhance your understanding of ethical considerations in cybersecurity.
+
+### Resources
+
+Some useful resources:
+
+- [Penetration testing](https://en.wikipedia.org/wiki/Penetration_test)
+- [Dirsearch](https://www.kali.org/tools/dirsearch/)
+- [Port Scanning](https://en.wikipedia.org/wiki/Port_scanner)
+- [HTTP headers](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers)
+
+Before asking for help, ask yourself if you have really thought about all the possibilities.
+
+### Role play
+
+To enhance the learning experience and assess your knowledge, a role play question session will be included as part of this project. This section will involve answering a series of questions in a simulated real-world scenario where you assume the role of a Cyber Security Expert explaining how to use pentesting tools to a team or stakeholder.
+
+The goal of the role play question session is to:
+
+- Assess your understanding of pentesting tools and techniques.
+- Test your ability to communicate effectively and explain security measures related to this project.
+- Challenge you to think critically about the importance of penetration testing and consider alternative approaches.
+
+Prepare for a role play question session in the audit.
+
+### Project Requirements
+
+#### Tool Development:
+
+You will develop the following tools:
+
+1. **TinyScanner (Simple Port Scanner):**
+
+   - Create a tool that checks for open, closed, or filtered ports on a target host.
+   - The tool should support TCP scanning and the ability to scan a range of ports.
+
+2. **DirFinder (Small Dirsearch):**
+
+   - Develop a tool to discover hidden directories and files on a web server by brute-forcing common paths using a wordlist.
+
+3. **HostMapper:**
+
+   - Build a tool to perform a ping sweep to identify live hosts on a subnet.
+
+4. **HeaderGrabber:**
+   - Implement a tool that retrieves and analyzes HTTP headers from a web server.
```

Comment:
From this description, the provided examples, links and a quick Google search I didn't understand what should be the output of this tool. To clarify the expected output, maybe, add:
- a standard tool that can be used as a reference to check what should be the expected output
- more information about the expected results, such as: reporting any potential missing security headers

Reply:
based on the example and the description of the tool, the tool is clear.
so no need for more information to describe it.